elastic/beats
# [filebeat] Fix ingest pipeline overwriting module field values

## What does this PR do?

This PR fixes a problem with the ingest pipeline not fully considering the fields included in the module configuration.

## Notes

According to [Filebeat doc](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-general-options.html#libbeat-configuration-fields), users can add new fields to the output, but doesn't mention anything about overwriting log entry's existing field values 

## Checklist




## How to test this PR locally

- Start local Kibana and add logging config to `kibana.yml`
```
logging:
  appenders:
    console:
      type: console
      layout:
        type: pattern
        highlight: true
    file:
      type: file
      fileName: ./logs/kibana-json.log
      layout:
        type: json
  root:
    appenders: [default, console, file]
    level: debug
```

Pull this branch and start filebeat from the source https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/dev_docs/how_to/running_components_from_source.md#filebeat

- On `filebeat.yml`, enable Kibana module.
```
- module: kibana
    log:
      enabled: true
      var.paths:
        - PATH_TO_KIBANA_LOG
      input:
        fields:
          ecs.version: "9.0.0" # existing field
          log.level: "TEST" # existing field
          service.name: "Kibana" # new field
          cloud.availability_zone: "danger-zone" # new field
        fields_under_root: true
```
Note that if `fields_under_root` is omitted or `false`, these custom fields will appear on the log as `fields.ecs.version`, `fields.kibana.service` and etc. When `true`, besides being included in the root, they can overwrite existing log entry fields.

<img width="1709" alt="image" src="https://user-images.githubusercontent.com/2767137/193795673-cc1777ac-955d-41a7-a624-8b39b92c7c71.png">

`cloud.availability_zone` and `service.name` will be added to the ingested log, `service.*` won't be overwritten by `service.name` and `ecs.version` and `log.level` will remain with their original value

## Related issues

Closes [#32665](https://github.com/elastic/beats/issues/32665)
